app/controllers/picture_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PictureController < ApplicationController
+  def show # GET /pictures/:basename
+    compressed_id, extension = params[:basename].split('.')
+    picture = Picture.find_by_id(Picture.uncompress_id(compressed_id))
+    if picture.blank? || extension != picture.extension
```

Comment:
`blank?` is a weird choice here.  I would think a nil check would make the most sense  Also the inverted negative logic is just so confusing

```ruby
if picture && picture.extension == extension
  render_picture_content(picture)
else
  render :nothing => true, :status => 404
end
```